jax/_src/interpreters/batching.py

```diff
@@ -98,23 +98,28 @@ def _pile_unflatten(aval, x):
   return Pile(aval, data)
 register_pytree_node(Pile, _pile_flatten, _pile_unflatten)
 
-def _pile_result(axis_size, axis, segment_lens, x):
+def _pile_result(axis_size, stacked_axis, ragged_axis, segment_lens, x):
   binder = core.Var(0, '', core.ShapedArray((), np.dtype('int32')))
+  if stacked_axis != 0:
+    raise NotImplementedError  # TODO Transpose x so the stacked axis is axis 0
   shape = list(x.shape)
-  shape[axis] = IndexedAxisSize(binder, segment_lens)
+  del shape[0]
+  shape[ragged_axis-1] = IndexedAxisSize(binder, segment_lens)
   elt_ty = core.DShapedArray(tuple(shape), x.dtype, x.weak_type)
   return Pile(PileTy(binder, axis_size, elt_ty), x)
 
+
 @dataclasses.dataclass(frozen=True)
-class ConcatAxis:
-  axis: int
+class RaggedAxis:
```

Comment:
Could you leave a TODO here, or in the PR message, explaining how we need to generalize `RaggedAxis` further, to handle piles like `i:(Fin 3) => f32[lens1.i, lens2.i]` (i.e. multiple ragged axes)?

Reply:
Done.